java-sdk/src/main/java/org/sagebionetworks/bridge/sdk/ClientProvider.java

```diff
@@ -30,44 +31,51 @@
     public static Config getConfig() {
         return config;
     }
-    
+
     public static ClientInfo getClientInfo() {
         return info;
     }
-    
+
     public static synchronized void setClientInfo(ClientInfo clientInfo) {
-        info = checkNotNull(clientInfo); 
+        info = checkNotNull(clientInfo);
     }
-    
+
     public static void addLanguage(String language) {
         checkArgument(isNotBlank(language));
-        
+
         // Would like to parse and verify this as a LanguageRange object,
         // which fully supports Accept-Language header syntax. It's a Java 8
         // feature and we're on Java 7. Put it on the TODO list.
-        languages.add(language);    
+        languages.add(language);
     }
-    
+
     public static LinkedHashSet<String> getLanguages() {
         return Utilities.newLinkedHashSet(languages);
     }
-    
+
     public static void clearLanguages() {
         languages = new LinkedHashSet<>();
     }
 
     /**
      * Sign In to Bridge with the given credentials. Returns a session object to manipulate. Method is idempotent.
      *
-     * @param signIn
+     * @param signInCredentials
      *            The credentials you wish to sign in with.
      * @return Session
      */
-    public static Session signIn(SignInCredentials signIn) throws ConsentRequiredException {
-        checkNotNull(signIn, "SignInCredentials required.");
+    public static Session signIn(SignInCredentials signInCredentials) throws
+            ConsentRequiredException {
+        checkNotNull(signInCredentials, "SignInCredentials required.");
 
-        UserSession userSession = new BaseApiCaller(null).post(config.getSignInApi(), signIn, UserSession.class);
-        return new BridgeSession(userSession);
+        SignIn signIn = new SignIn().study(signInCredentials.getStudyIdentifier())
+                                    .email(signInCredentials.getEmail())
+                                    .password(signInCredentials.getPassword());
```

Comment:
If I throws this into a file and format it, I get something more like this:

        SignIn signIn = new SignIn().study(signInCredentials.getStudyIdentifier()).email(signInCredentials.getEmail())
                .password(signInCredentials.getPassword());

Sometimes I align this to the first indent, so more like this:

        SignIn signIn = new SignIn()
                .study(signInCredentials.getStudyIdentifier())
                .email(signInCredentials.getEmail())
                .password(signInCredentials.getPassword());